analysis/thread-dump/src/main/java/org/eclipse/jifa/tda/ThreadDumpAnalyzer.java

```diff
@@ -305,4 +297,31 @@ public Map<MonitorState, Integer> threadCountsByMonitor(int id) {
         map.forEach((s, l) -> counts.put(s, l.size()));
         return counts;
     }
+
+    /**
+     * @return rows of a thread dump table
+     */
+     public List<ThreadDumpRow> rows(PagingRequest paging) {
+
+         List<ThreadDumpRow> threadDumpRows = new ArrayList<>(snapshot.getJavaThreads().stream().map(t -> {
+
+             Frame[] frames = null;
+             int length = 0;
+
+             if (t.getTrace() != null) {
+                 frames = t.getTrace().getFrames();
+                 length = frames.length;
+             }
+
+             return new ThreadDumpRow(t.getName(), t.getId(), t.getOsThreadState(), length, frames, t.getElapsed(), t.getCpu(), t.getLineStart(), t.getLineEnd());
+         }).toList());
+
+         List<ThreadDumpRow> nonJavaThreadDumpRows = snapshot.getNonJavaThreads().stream().map(t -> {
+             return new ThreadDumpRow(t.getName(), t.getId(), t.getOsThreadState(), 0, new Frame[]{}, t.getElapsed(), t.getCpu(), t.getLineStart(), t.getLineEnd());
+         }).toList();
+
+         threadDumpRows.addAll(nonJavaThreadDumpRows);
```

Comment:
This approach generates quite a bit of intermediary list garbage. Consider something like this instead:

```
Stream<ThreadDumpRow> javaRows = snapshot.getJavaThreads().stream().map(....);
Stream<ThreadDumpRow> nonJavaRows = snapshot.getNonJavaThreads().stream().map(....);
Stream<ThreadDumpRow> allRows = Stream.concat(javaRows, nonJavaRows);
// consider implementing paging here on the stream
List<ThreadRumpRow> threadDumpRows = allRows.toList();
```